File: app/src/main/java/com/yoneya1014/jetfarm_test1/SavedDataSelectMenuActivity.java
package com.yoneya1014.jetfarm_test1;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

public class SavedDataSelectMenuActivity extends AppCompatActivity {

    public static Boolean deleteFrag = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.saveddataselectmenu);
        ListView listView = findViewById(R.id.listView);
        ArrayAdapter adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, MainMenuActivity.datalist);
        listView.setAdapter(adapter);
        setTitle("登録データ一覧");
        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                ListView list = (ListView) parent;
                String text = (String) list.getItemAtPosition(position);
                Intent intent = new Intent(getApplicationContext(), ShowSavedDataActivity.class);
                intent.putExtra("dataname", text);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onResume() {
        super.onResume();
        if (deleteFrag) {
            deleteFrag = false;
            finish();
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/yoneya1014/jetfarm_test1/SavedDataSelectMenuActivity.kt
package com.yoneya1014.jetfarm_test1

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.firestore.FirebaseFirestore
import java.text.SimpleDateFormat
import java.util.*

class SavedDataSelectMenuActivity : AppCompatActivity() {

    private var dataList: MutableList<String> = ArrayList()
    private var listView: ListView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        dataList.clear()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.saveddataselectmenu)
        listView = findViewById(R.id.listView)
    }

    public override fun onResume() {
        super.onResume()
        val progressDialog = ProgressDialog(this)
        progressDialog.setMessage("リストを更新中です")
        progressDialog.setCancelable(false)
        progressDialog.isIndeterminate = false
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER)
        progressDialog.show()
        dataList.clear()
        FirebaseFirestore.getInstance().collection("userSavedData")
                .get()
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        for (document in task.result!!) {
                            val date = document.getTimestamp("timestamp")!!.toDate()
                            val format = SimpleDateFormat("yyyy年MM月dd日HH時mm分ss秒", Locale.JAPAN)
                            dataList.add(format.format(date))
                        }
                        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, dataList)
                        listView!!.adapter = adapter
                        title = "登録データ一覧"
                        val actionBar = supportActionBar
                        actionBar!!.setDisplayHomeAsUpEnabled(true)
                        listView!!.onItemClickListener = AdapterView.OnItemClickListener { parent, _, position, _ ->
                            val list = parent as ListView
                            val text = list.getItemAtPosition(position) as String
                            val intent = Intent(applicationContext, ShowSavedDataActivity::class.java)
                            intent.putExtra("dataName", text)
                            startActivity(intent)
                        }
                        progressDialog.dismiss()
                    } else {
                        Log.d("Result", "Error")
                    }
                }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
